### PR TITLE
Add-Address Identicon Selection

### LINKED ISSUE #220: Move add-address flow into Accounts
The add-address flow should include an overlay (after the click on an identicon), which already includes an input field to set the new address's label. This input field should default to the IqonName for that address.

Since the Keyguard does not (and should not) include the IqonName library (because of code duplication and unnecessary complexity in the Keyguard), this issue outlines a proposal to move the identicon selection and naming into the Accounts Manager:

Instead of selecting the identicon in the Keyguard, the Keyguard is only used to decrypt the key with the password. Then it returns to Accounts to display the identicon selector and naming overlay there.

There are two ways this can go:

1. Decrypt the key in the Keyguard and store it in session storage, just like during login/import. The AM would then request derived addresses through a Keyguard iframe.
2. The AM already sends the list of address to derive (`indicesToDerive`) into the Keyguard, so the Keyguard could just return _all_ of those addresses (and their paths if necessary) to the AM, which then does the identicon selection and naming.

Option 2 is favored, as it would not store an unencrypted key in the session storage and the number of addresses to derive is known at the beginning of the request anyway. Furthermore, the identicon selector already exists in Accounts because Ledger flows also use it.

### PR DESCRIPTION
**Pull request by sisou**
Uses https://github.com/nimiq/keyguard-next/pull/352 for updated result types.

Resolves #220.

Moves the identicon selector for the add-address request into the Hub, to be able to use show the user a label input and suggest auto-labelling.

### CODE REVIEW

File: src/components/IdenticonSelector.vue [hunk 1]
```diff
@@ -196,7 +194,7 @@
         left: 0;
         height: 100%;
         width: 100%;
-        padding: 4rem;
+        padding: 4rem 3rem;
```
Comment (nibhar) [inline src/components/IdenticonSelector.vue]:
```suggestion
        padding: 4rem 3rem;
        border-radius: 1rem;
```

File: src/views/AddAccountSelection.vue [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        if (!selectedAccount.label) {
+            selectedAccount.label = selectedAccount.defaultLabel;
+        }
+
+        wallet.accounts.set(selectedAccount.userFriendlyAddress, selectedAccount);
+
+        await WalletStore.Instance.put(wallet);
+        // Artificially delay, to display loading status
+        await new Promise((res) => setTimeout(res, 1000));
```
Comment (danimoh) [inline src/views/AddAccountSelection.vue]: Why display the loading status at all?
Display the success screen immediately after the wallet was stored and don't even go into the loading state as storing the wallet is very quick.

File: src/views/AddAccountSelection.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    private get accounts(): AccountInfo[] {
+        return this.keyguardResult.map((address) => new AccountInfo(
+            address.keyPath,
+            '',
+            new Nimiq.Address(address.address),
+        ));
+    }
+
+    private back() {
```
Comment (danimoh) [inline src/views/AddAccountSelection.vue]: Private methods should start with an underscore....

File: src/components/IdenticonSelector.vue [hunk 4]
```diff
@@ -1,7 +1,9 @@
 <template>
     <div class="identicon-selector" :class="{ 'account-details-shown': selectedAccount && confirmAccountSelection }"
         @keydown.esc="_selectAccount(null)">
-        <h1 class="nq-h1 blur-target">Choose an Avatar</h1>
+        <div class="blur-target">
+            <slot></slot>
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/components/IdenticonSelector.vue]: If using a named slot, using components have to wrap their content in an additional `<template slot="header">` tag.

File: src/components/IdenticonSelector.vue [hunk 5]
```diff
@@ -23,14 +25,14 @@
         <transition name="transition-fade">
             <div v-if="selectedAccount && confirmAccountSelection" @click="_selectAccount(null)"
                 class="account-details">
+                <button @click="_selectAccount(null)" class="nq-button-s close-button">
+                    <CloseIcon/>
+                </button>
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/components/IdenticonSelector.vue]: That's also my argument. The button is visually above the input, so I expect tabbing to go there first before the input.

File: src/views/AddAccountSelection.vue [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        <SmallPage>
+            <IdenticonSelector  v-if="!showLoader" :accounts="accounts" @identicon-selected="identiconSelected">
+                <PageHeader backArrow @back="back">Choose a new Account</PageHeader>
+            </IdenticonSelector>
+
+            <Loader v-else :title="title" :state="state" :lightBlue="true">
+                <template slot="success">
+                    <CheckmarkIcon/>
+                    <h1 class="title nq-h1">New address added<br>to your account.</h1>
+                </template>
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/views/AddAccountSelection.vue]: Just specifying the title does not allow for manual line breaks.

File: src/views/AddAccountSelection.vue (hunk 2 above)
Comment (sisou) [reply src/views/AddAccountSelection.vue]: Except when it isn't and the database is blocked for whatever reason.

File: src/components/IdenticonSelector.vue (hunk 1 above)
Comment (sisou) [reply src/components/IdenticonSelector.vue]: @danimoh You have, but it has no effect (anymore, maybe).

File: src/components/IdenticonSelector.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (sisou) [reply src/components/IdenticonSelector.vue]: Ok, the `<template>` tag is not required apparently.

File: src/views/AddAccountSelection.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (danimoh) [reply src/views/AddAccountSelection.vue]: You can have the line break in the provided string and set the `white-space` in `Loader` accordingly.

File: src/views/AddAccountSelection.vue (hunk 2 above)
Comment (danimoh) [reply src/views/AddAccountSelection.vue]: I think it's unlikely that that happens. Not worth letting all users wait 1s.
If you really want handle the case of a long pending database transaction, check after a short delay (maybe 100ms) whether the wallet was stored and only if not display the loading state.

File: src/views/AddAccountSelection.vue (hunk 3 above)
Comment (danimoh) [reply src/views/AddAccountSelection.vue]: We had a decision on that.

File: src/views/AddAccountSelection.vue [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            </Loader>
```
[the remark this answers is not in the document]
Comment (danimoh) [reply src/views/AddAccountSelection.vue]: No state transitions added.

File: src/components/IdenticonSelector.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (danimoh) [reply src/components/IdenticonSelector.vue]: I think prioritizing the element the user is likely to click provides the better UI.

**Comment by danimoh** [comment]
Don't mark issues as resolved that are actually not resolved. 
Most of my change requests have not been addressed accordingly.
Please fix.

File: src/components/IdenticonSelector.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (sisou) [reply src/components/IdenticonSelector.vue]: I think tabbing in the order of the elements from top to bottom in the page is better UX.

@julianbauer ?

File: src/views/AddAccountSelection.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (sisou) [reply src/views/AddAccountSelection.vue]: You are welcome to open an issue for this, as other success screens/loaders do the same, so you can change it in all places at once.

File: src/views/AddAccountSelection.vue (hunk 3 above)
Comment (sisou) [reply src/views/AddAccountSelection.vue]: There are right now 10 `private _*` occurances vs 103 `private`s without `_` in `vue-components/src` and 13 `private _*` occurances vs 177 `private`s without `_` in `hub/src/*/*.vue` files.

You are welcome to also open an issue or PR to change all of them, but the stats are currently 10:1 in the non-underscore favor.

Comment (danimoh) [reply src/views/AddAccountSelection.vue]: Alright, I'll change it some day throughout the code base.

File: src/views/AddAccountSelection.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (danimoh) [reply src/views/AddAccountSelection.vue]: Opened this issue: https://github.com/nimiq/hub/issues/290

**Pull request merged**